### DISCUSSION TITLE
Separate modules development

### DISCUSSION THREAD

**Discussion by zombozo12** [nWidart/laravel-modules#1320] [Q&A]
Hi, Is it possible to develop each modules separately and how? I found about `Publishing Modules` in the documentation, but I don't quite understand how I push them to the repository. Should I push the whole project includes the module or I just need to push on the module directory?

References:
[Publishing Module](https://nwidart.com/laravel-modules/v6/advanced-tools/publishing-modules)

**Comment by maxcelos** [comment]
Only the specific module you want to publish.

**Reply by zombozo12** [reply]
Thank you for your reply. Can you please elaborate more information? Is it only 1 module or all of my modules? If we develop all modules inside a main laravel projects, our code editor might get slow or we as developers might confused by many files inside the main project which contains bunch of modules.

**Reply by maxcelos** [reply]
I see your concern. I didn't run into an issue like that but a simpler solution I would try is to have a main project with the dependencies to run any module. So your developers can clone this project and start to implement the module as a "git submodule".

In my workflow, I try to export all base modules and install them through composer. This way they go do vendor folder and I can focus in the modules for that particular project only.

I have a "foundation-module" that sets up all. So if I want to start a new project, I just create a new laravel project from scratch, install this module, ran a setup command, and I'm ready to start creating my new modules.
